File: php/testcode/hardcodedcreds_024.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_hardcoded_base64_cred
function hardcodedcreds024(BenchmarkRequest $req): BenchmarkResponse {
    $dbPass = base64_decode('aHVudGVyMg=='); // vuln-code-snippet vuln-line php_hardcoded_base64_cred
    $pdo = new PDO('mysql:host=localhost;dbname=app', 'root', $dbPass);
    $stmt = $pdo->query('SELECT 1');
    return BenchmarkResponse::ok('connected');
}
// vuln-code-snippet end php_hardcoded_base64_cred

File: rust/testcode/hardcodedcreds_024.rs
//! CWE-798: Hardcoded admin password compared directly against user input.

// vuln-code-snippet start testcodeHardcodedcreds024
const ADMIN_PASSWORD: &str = "Admin@123!";

pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let password = req.param("password");
    if password == ADMIN_PASSWORD { // vuln-code-snippet target-line testcodeHardcodedcreds024
        super::shared::BenchmarkResponse::ok("Login successful")
    } else {
        super::shared::BenchmarkResponse::forbidden("Invalid password")
    }
}
// vuln-code-snippet end testcodeHardcodedcreds024
